### PR TITLE
Add support for lookup requests

### PR DESCRIPTION
**Pull request by andreaspsson**
Now we can fetch multiple items by id on one request, this can be useful when building join-like things

### CODE REVIEW

File: src/query.rs [hunk 1]
```diff
@@ -355,6 +364,48 @@ async fn get_one_by_id(
...
+                    Ok(result)
+                })
+                .collect::<Result<Vec<DatastoreEntity>, DatastorersError>>()?;
+            Ok(result)
+        }
+        None => Err(DatastoreClientError::NotFound.into()),
+    }
+}
+
+async fn do_lookup(
```
Comment (bes) [inline src/query.rs]: raw_lookup ?

File: src/query.rs [hunk 2]
```diff
@@ -355,6 +364,48 @@ async fn get_one_by_id(
     }
 }
 
+async fn lookup(
```
Comment (bes) [inline src/query.rs]: entity_lookup ?

File: src/query.rs [hunk 3]
```diff
@@ -12,13 +14,14 @@ use crate::identifier::KeyPath;
...
     PropertyFilter, PropertyFilterOp, PropertyOrder, PropertyOrderDirection, PropertyReference,
     Query, QueryResultBatchMoreResults, ReadOptions, RunQueryRequest, RunQueryResponse,
 };
 
 const DEFAULT_PAGE_SIZE: i32 = 50;
 
+#[async_trait]
 pub trait DatastorersQueryable<E>
 where
     E: Kind
```
Comment (bes) [inline src/query.rs:27]: Why does `DatastorersQueryable<E>`  require `E` to implement `DatastorersQueryable<E>` seems a bit recursive/redundant?

**Review by bes — APPROVED** [review]

File: src/query.rs (hunk 1 above)
Comment (andreaspsson) [reply src/query.rs]: 👍

File: src/query.rs (hunk 2 above)
Comment (andreaspsson) [reply src/query.rs]: 👍

File: src/query.rs (hunk 3 above)
Comment (andreaspsson) [reply src/query.rs:27]: That was a really good question!
looks kind of strange, any you know what, with some minor changes the requirement could be removed.
Included fix in this PR, see commit: 
https://github.com/et-alia/datastorers/pull/52/commits/4c6a04b4d6ab7a5ef82466314a7fa2de60c121cd

**Pull request merged**